outlines/function.py

```diff
@@ -10,6 +10,9 @@
     from outlines.generate.api import SequenceGenerator
     from outlines.prompts import Prompt
 
+# Raising a warning here caused all the tests to fail…
+print("The 'function' module is deprecated and will be removed in a future release.")
+
```

Comment:
Maybe better place would be putting warning in `__post_init__` of `Function`?

Reply:
Moving my previous comment here for visibility:

> I would be nice to specify which release (1.0.0) so people can pin their version to < this release?